cli/cmd/upgrade.go

```diff
@@ -96,18 +127,12 @@ install command.`,
 		},
 	}
 
-	// add this flag directly rather than as part of the FlagSet because we do not
-	// want it persisted into linkerd-config/install ConfigMap
-	cmd.PersistentFlags().StringVar(
-		&options.manifests, "from-manifests", options.manifests,
-		"Read config from a Linkerd install YAML rather than from Kubernetes",
-	)
-
 	cmd.PersistentFlags().AddFlagSet(flags)
+	cmd.PersistentFlags().AddFlagSet(upgradeOnlyFlags)
 	return cmd
 }
 
-func (options *upgradeOptions) validateAndBuild(k kubernetes.Interface, flags *pflag.FlagSet) (*installValues, *pb.All, error) {
+func (options *upgradeOptions) validateAndBuild(stage string, k kubernetes.Interface, flags *pflag.FlagSet) (*installValues, *pb.All, error) {
```

Comment:
I don't think `stage` is needed in this function; should be able to assign it to `values` after the call to `validateAndBuild()`.

Reply:
I considered this initially as it would have minimized the change and preserved `validateAndBuild` signature. I opted to set `stage` in this function keep all the code that creates an `installValues` in one place. The alternative is that `validateAndBuild` creates a nearly-complete `installValues` that the caller must know to set `stage` after the fact.